app/controllers/accounts_controller.rb

```diff
@@ -19,6 +19,13 @@ def show
       user: current_user,
       locked_for_session: pii_locked_for_session?(current_user),
     )
+    if session[:from_select_email_flow]
+      flash[:success] = t(
+        'account.emails.confirmed_html',
+        url: account_connected_accounts_url,
+      )
+      session.delete(:from_select_email_flow)
```

Comment:
Minor: Dunno if you have any feelings on readability, but could do the read & delete in a single operation if wanted.

```suggestion
    if session.delete(:from_select_email_flow)
      flash[:success] = t(
        'account.emails.confirmed_html',
        url: account_connected_accounts_url,
      )
```